Mathlib/LinearAlgebra/LinearPMap.lean

```diff
@@ -1005,11 +1033,161 @@ theorem mem_inverse_graph (x : f.domain) : (f x, (x : E)) ∈ (inverse f).graph
     exists_eq_left, LinearEquiv.prodComm_apply, Prod.exists, Prod.swap_prod_mk, Prod.mk.injEq]
   exact ⟨(x : E), f x, ⟨x.2, Eq.refl _⟩, Eq.refl _, Eq.refl _⟩
 
+theorem mem_graph_of_inverse (y : f.inverse.domain) : (f.inverse y, (y : F)) ∈ f.graph := by
+  simp [graph_eq_inverse_graph_prodComm hf]
+
 theorem inverse_apply_eq {y : (inverse f).domain} {x : f.domain} (hxy : f x = y) :
     (inverse f) y = x := by
   have := mem_inverse_graph hf x
   grind [mem_graph_iff]
 
+end injective
+
+section bijective
+
+/-- The domain of the inverse of `f` is the full space if and only if `f` is bijective. -/
+theorem inverse_domain_eq_top_iff : f.inverse.domain = ⊤ ↔ LinearMap.range f.toFun = ⊤ := by
+  rw [inverse_domain]
+
+open Classical in
+/-- If `f` is bijective, then the inverse is defined as a linear map. -/
+noncomputable def inverse_asLinearMap (g : E →ₗ.[R] F) : F →ₗ[R] E :=
+  if hg : LinearMap.range g.toFun = ⊤ then g.inverse.toFun.comp (LinearEquiv.ofTop g.inverse.domain
+    ((inverse_domain_eq_top_iff).mpr hg)).symm.toLinearMap
+    else 0
```

Comment:
I'm not yet convinced that we want the junk value approach *everywhere*. In particular, if you're doing this upgrade, I would expect you probably aren't going to be rewriting it much.